DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/CompetencyDataService.cs

```diff
@@ -423,15 +406,9 @@ UPDATE SelfAssessmentResults
                                 [SupportingComments] = @supportingComments
                             WHERE ID = @existentResultId
                         ELSE
-                            INSERT INTO SelfAssessmentResults
-                                ([SelfAssessmentID]
-                                ,[CompetencyID]
-                                ,[AssessmentQuestionID]
-                                ,[Result]
-                                ,[DateTime]
-                                ,[SupportingComments]
-                                ,[DelegateUserID])
-                            VALUES(@selfAssessmentId, @competencyId, @assessmentQuestionId, @result, GETUTCDATE(), @supportingComments,@delegateUserId)
+                            UPDATE SelfAssessmentResults

```

Comment:
If we update a `SelfAssessmentResults` record we need to delete any related `SelfAssessmentResultSupervisorVerifications` record.